src/init.cpp

```diff
@@ -632,6 +633,11 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     // ********************************************************* Step 0: Load zcash params
     ZC_LoadParams();
 
+    // Perform libsodium initialization
+    if (sodium_init() == -1) {
```

Comment:
I think that's fair. `ZC_LoadParams` won't do anything with our PRNG even if we add hooks, but it would be more sensible to reorder these.